airbyte-integrations/connectors/source-my-hours/source_my_hours/source.py

```diff
@@ -0,0 +1,152 @@
+
+
+class MyHoursStream(HttpStream, ABC):
+    url_base = URL_BASE + "/"
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        return None
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        for record in response.json():
+            yield record
+
+    def request_headers(
+        self, stream_state: Mapping[str, Any], stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> Mapping[str, Any]:
+        return REQUEST_HEADERS
+
+
+class Clients(MyHoursStream):
+    primary_key = "id"
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "Clients"
+
+
+class Projects(MyHoursStream):
+    primary_key = "id"
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "Projects/getAll"
+
+
+class Tags(MyHoursStream):
+    primary_key = "id"
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "Tags"
+
+
+class TimeLogs(MyHoursStream):
+    primary_key = "logId"
+
+    def __init__(
+        self,
+        authenticator: TokenAuthenticator,
+        start_date: str,
+        batch_size: int,
+        **kwargs,
+    ):
+        super().__init__(authenticator=authenticator)
+
+        self.start_date = pendulum.parse(start_date)
+        self.batch_size = batch_size
+
+        if self.start_date > pendulum.now():
+            self.logger.warn(f'Stream {self.name}: start_date "{start_date.isoformat()}" should be before today.')
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return "Reports/activity"
+
+    def next_page_token(self, response: requests.Response) -> Optional[Mapping[str, Any]]:
+        previous_query = parse_qs(urlparse(response.request.url).query)
+        previous_end = pendulum.parse(previous_query["DateTo"][0])
+
+        new_from = previous_end.add(days=1)
+        new_to = new_from.add(days=self.batch_size - 1)
+
+        if new_from > pendulum.now():
+            return None
+
+        return {
+            "DateFrom": new_from.to_date_string(),
+            "DateTo": new_to.to_date_string(),
+        }
+
+    def request_params(
+        self,
+        stream_state: Mapping[str, Any],
+        stream_slice: Mapping[str, Any] = None,
+        next_page_token: Mapping[str, Any] = None,
+    ) -> MutableMapping[str, Any]:
+
+        if next_page_token is None:
+            return {"DateFrom": self.start_date.to_date_string(), "DateTo": self.start_date.add(days=6).to_date_string()}
```

Comment:
```suggestion
            return {"DateFrom": self.start_date.to_date_string(), "DateTo": self.start_date.add(days=self.batch_size).to_date_string()}
```

Reply:
Changed this to be `days=self.batch_size - 1` similar to `next_page_token`